design-proposals/memory-hotplug.md

```diff
@@ -0,0 +1,209 @@
+spec:
+  domain:
+    memory:
++     maxGuest: 10Gi
+  guest: 8Gi
+
+```
+
+## Status API
+
+The VMI status must indicate whether Memory hotplug is taking place. As the hotplug process begins the current state
+of `spec.domain.memory` needs to be propagated back to the VM/VMI `status.currentMemory` to allow the user to have an
+indication of the hotplug processing.
+
+```
+	// Memory allow specified the detailed inside the vmi.
+		CurrentMemory *Memory `json:"currentMemory,omitempty"`
+```
+
+Conditions on the VM and VMI should indicate the status of the pending Memory hotplug actions. When errors occur, the
+conditions should propagate the errors to the VM.Status.Conditions field so users can receive feedback into the status
+of their declared state.
+
+```
+// Indicates that the VMI is in progress of Hot memory Plug/UnPlug
+VirtualMachineInstanceMemoryChange VirtualMachineInstanceConditionType = "HotMemoryChange"
+```
+
+Upon succesful hotplug action the condition will be removed. Otherwise, a human readable reason and message indicating
+whether the hotplug change is pending or encountering errors.
+
+## Extensibility
+
+We will be able to use the `VMLiveUpdateFeatures` section to manage things like cpu/memory hotplug .
+
+## Update/Rollback Compatibility
+
+The feautre doesn't intorduce any breaking changes so existing functionality of versions prior to this feature should
+not be impacted upon update/rollback. However if existing workloads would like to opt-in for this feature after an
+update, the VMI should be re-created since the addition of new fields to the VMI API.
+
+## Functional Testing Approach
+
+Functional tests will simply:
+
+* Create and start a VM with `memeory` set as one of th `liveUpdateFeatures`
+* Modify the number of memory guest for the VM
+* Ensure the guest sees the changed number of memory
+* Ensure the memory resources on the virt-launcher pod were adjusted accordingly
+* Ensure interoperability with non-hotplug related live-migration occasions (workload-update, evacuation,
+  user-triggered)
+
+## Implementation Phases
+
+### Phase 1: Creating the KubeVirt API and linking it to the LibVirt memory hotplug API
+
+### Phase 2: Triggering a live migration when the number of sockets changes, to adjust the virt-launcher memory resources
+
+The
+VMI [workload update controller](https://github.com/kubevirt/kubevirt/blob/ce8aab7874d2e1586787e3e2a17306b7edca1b8a/pkg/virt-controller/watch/workload-updater/workload-updater.go#L476)
+is extended to live migrate VMIs to satisfy updates to a VMI resource request/limits field.
+
+Sequence of events:
+
+* User updates memory guest on a vm.spec.template.spec.domain.memory
+* VM controller determines the request/limits required and writes the necessary changes to the active vmi.Spec
+* VM controller writes the required request/limits in the vmi.Status
+* VMI workload update controller detects the mismatch between the resource request/limits on the pod vs vmi.Spec and
```

Comment:
Can you rewrite this section? Most things are correct but the actual sequence is described in this commit https://github.com/kubevirt/kubevirt/pull/10184/commits/d0350f2e7bb1af2d5834d1fc146a514dde18e786.